toolchain/lower/file_context.cpp

```diff
@@ -136,17 +137,21 @@ auto FileContext::BuildFunctionDecl(SemIR::FunctionId function_id)
     return nullptr;
   }
 
-  // Don't lower unused functions.
-  if (function.return_slot == SemIR::Function::ReturnSlot::NotComputed) {
-    return nullptr;
-  }
+  // TODO: Consider tracking whether the function has been used, and only
+  // lowering it if it's needed.
+
+  // TODO: Pass in a specific ID for generic functions.
+  const auto specific_id = SemIR::GenericInstanceId::Invalid;
 
-  const bool has_return_slot = function.has_return_slot();
+  const auto return_info = function.GetReturnInfo(sem_ir(), specific_id);
+  CARBON_CHECK(return_info.is_valid()) << "Should not lower invalid functions.";
+
+  const bool has_return_slot = return_info.has_return_slot();
```

Comment:
FWIW, I'd lean towards eliminating the `has_return_slot` local in spots like this. I'm not sure the locals in this file were needed before (by contrast, `bool has_return_slot = GetInitRepr(sem_ir_, inst.type_id).has_return_slot();` makes sense). But when you already have the `return_info` object, also having `return_info.has_return_slot()` separate feels like it's just being verbose.

Reply:
Done.